### PR TITLE
feat: Article Ci de qualité

### PR DESCRIPTION
**Pull request by CaptainJojo**
## description

DONE

### CODE REVIEW

**Comment by ElevenWilson** [comment]
38 words might be misspelled, please check them: airbnb, BackstopJs, Bithound, captainjojo, config, connaitre, CRON, css, CSS, déja, editorconfig, Editorconfig, Eslint, frontend, html, l'editorconfig, L'editorconfig, ok, phpunit, plugue, pre, psr, PSR, repo, répo, Scrutinizer, Speedcurve, Stylelint, stylelintrc, symfony, Symfony, sympas, technos, travis, trollLand, tutoriel, webperformance, WebPerformance

**Comment by ElevenWilson** [comment]
38 words might be misspelled, please check them: airbnb, BackstopJs, Bithound, captainjojo, config, connaitre, CRON, css, CSS, déja, editorconfig, Editorconfig, Eslint, frontend, html, l'editorconfig, L'editorconfig, ok, phpunit, plugue, pre, psr, PSR, repo, répo, Scrutinizer, Speedcurve, Stylelint, stylelintrc, symfony, Symfony, sympas, technos, travis, trollLand, tutoriel, webperformance, WebPerformance

**Comment by ElevenWilson** [comment]
36 words might be misspelled, please check them: airbnb, BackstopJs, Bithound, captainjojo, config, CRON, css, CSS, editorconfig, Editorconfig, Eslint, frontend, html, l'editorconfig, L'editorconfig, ok, phpunit, plugue, pre, psr, PSR, repo, répo, Scrutinizer, Speedcurve, Stylelint, stylelintrc, symfony, Symfony, sympas, technos, travis, trollLand, tutoriel, webperformance, WebPerformance

**Comment by VEBERArnaud** [comment]
Cool l'article, ca rappel de bons souvenirs !

Je t'ai commenté 2-3 trucs qui peuvent être corrigés rapidos, sinon Kudos !

**Comment by CaptainJojo** [comment]
@VEBERArnaud done merci. Et oui plein de bon souvenirs :)

**Comment by ElevenWilson** [comment]
36 words might be misspelled, please check them: airbnb, BackstopJs, Bithound, captainjojo, config, CRON, css, CSS, editorconfig, Editorconfig, Eslint, frontend, html, l'editorconfig, L'editorconfig, ok, phpunit, plugue, pre, psr, PSR, repo, répo, Scrutinizer, Speedcurve, Stylelint, stylelintrc, symfony, Symfony, sympas, technos, travis, trollLand, tutoriel, webperformance, WebPerformance

File: _posts/2017-08-30-votre-ci-de-qualite.md [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+
+La qualité est un vaste sujet, surtout quand on l'associe au développement d'application web.
+
+Ce qui est encore plus compliqué, c'est de mettre en place l'environnement d'intégration continue (CI) de suivi de qualité.
+
+Pendant plus de 2 ans, nous avons mis en place une CI de qualité au Monde.fr qui a évolué en fonction de nos besoins. Le but de cet article est de comprendre la stratégie et les technos choisies pour la CI d'un site comme le Monde.fr
+
+### La partie Symfony
+
+Comme pour tout language de programmation, la première chose que l'on veut vérifier, c'est la syntaxe. La première chose à mettre en place est donc un vérificateur de syntaxe en PHP.
```
Comment (eko) [inline _posts/2017-08-30-votre-ci-de-qualite.md]: `language` -> `langage`

File: _posts/2017-08-30-votre-ci-de-qualite.md [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+    - javascript
+    - webperformance
+cover: /assets/2017-08-30-votre-ci-de-qualite/cover.jpg
+---
+
+La qualité est un vaste sujet, surtout quand on l'associe au développement d'application web.
+
+Ce qui est encore plus compliqué, c'est de mettre en place l'environnement d'intégration continue (CI) de suivi de qualité.
+
+Pendant plus de 2 ans, nous avons mis en place une CI de qualité au Monde.fr qui a évolué en fonction de nos besoins. Le but de cet article est de comprendre la stratégie et les technos choisies pour la CI d'un site comme le Monde.fr
```
Comment (eko) [inline _posts/2017-08-30-votre-ci-de-qualite.md]: `qualité au Monde.fr` -> `qualité chez LeMonde.fr`
`comme le Monde.fr` -> `comme LeMonde.fr`

File: _posts/2017-08-30-votre-ci-de-qualite.md [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+    if [ $? -eq 0 ]; then
+      EXITCODE=1
+    fi
+
+  fi
+done
+
+if [ $EXITCODE -gt 0 ]; then
+  echo
+  echo 'Fix the above erros or use:'
```
Comment (eko) [inline _posts/2017-08-30-votre-ci-de-qualite.md]: `erros` -> `errors`

File: _posts/2017-08-30-votre-ci-de-qualite.md [hunk 4]
```diff
@@ -0,0 +1,625 @@
...
+  echo ' git commit --no-validate'
+  echo
+fi
+
+exit $EXITCODE
+```
+
+Si tout est ok, lors de chaque `commit` , le hook va vérifier la syntaxe php.
+
+Une fois cela validé, la suite logique est de faire que les développeurs codent tous avec les mêmes standards. Ce qui est bien, c'est que PHP à déjà des standards : les **[PSR](http://www.php-fig.org/psr/)**.
```
Comment (eko) [inline _posts/2017-08-30-votre-ci-de-qualite.md]: `à déjà` -> `a déjà`

File: _posts/2017-08-30-votre-ci-de-qualite.md [hunk 5]
```diff
@@ -0,0 +1,625 @@
...
+
+Faut-il que tous les développeurs les suivent, c'est assez simple en PHP. Nous avons ajouté dans notre hook de pre-commit la commande de vérification disponible dans cette article, *[vérifier la qualité du code](http://blog.eleven-labs.com/fr/verifier-la-qualite-du-code/)*
```
Comment (eko) [inline _posts/2017-08-30-votre-ci-de-qualite.md]: `Faut-il` -> `Encore faut-il` ?
`cette article` -> `cet article`

File: _posts/2017-08-30-votre-ci-de-qualite.md [hunk 6]
```diff
@@ -0,0 +1,625 @@
...
+```yml
+//.travis.yml
+script:
+  ## PHPUnit
+  - vendor/bin/phpunit
+```
+
+**Bravo vos tests sont dans la CI !!!**
+
+Après cette importante étape, nous avons cherché à savoir ce qui nous manquait. Nous avions des tests mais cela ne dit pas que notre code est de qualité mais seulement qu'il est fonctionnel. Pour améliorer la qualité du code, nous avons alors instauré l'obligation d'un relecture par deux autres développeurs de chaque pull request afin qu'elle soit validée. Nous avions la sensation que la qualité était meilleure car les gens posaient les bonnes questions:
```
Comment (eko) [inline _posts/2017-08-30-votre-ci-de-qualite.md]: `d'un relecture` -> `d'une relecture`

File: _posts/2017-08-30-votre-ci-de-qualite.md [hunk 7]
```diff
@@ -0,0 +1,625 @@
...
+
+ 1. pourquoi cette variable ?
+ 2. ton nom de fonction est étrange.
+ 3. tu pourrais utiliser cette fonction.
+ 4. j'ai déjà codé un truc ressemblant.
+ 5. etc...
+
+Mais il nous est arrivé que certaine pull request nous ait posé des problèmes. Des questions du type `for` ou `while` sont apparues dans les codes review et cela est devenu trollLand sur certaine pull request. Comment faire ?
```
Comment (eko) [inline _posts/2017-08-30-votre-ci-de-qualite.md]: `que certaine pull request nous ait posé des problèmes` -> `que certaines pull request nous aient posées des problèmes`

File: _posts/2017-08-30-votre-ci-de-qualite.md [hunk 8]
```diff
@@ -0,0 +1,625 @@
...
+On l'oublie souvent mais le CSS, c'est aussi du code, et la qualité de celui-ci doit aussi être prise en compte.
+
+Jamais deux sans trois, on commence par la syntaxe. Pour cela, nous avons utilisé [Stylelint](https://stylelint.io/) qui permet de gérer la syntaxe de vos fichiers CSS. Stylelint permet de nombreuses vérifications:
+
+ 1. Ne pas avoir de commentaire vide
+ 2. Le nombre de sélecteur max
+ 3. Vérification des accolades
+ 4. etc...
+
+Si vous suivez le tutoriel, vous devez écrire un fichier de configuration à la racine de votre répo.
```
Comment (eko) [inline _posts/2017-08-30-votre-ci-de-qualite.md]: `répo` -> `repo(sitory?)`

**Comment by CaptainJojo** [comment]
@eko done :) merci beaucoup.

**Comment by ElevenWilson** [comment]
37 words might be misspelled, please check them: airbnb, BackstopJs, Bithound, captainjojo, config, CRON, css, CSS, editorconfig, Editorconfig, Eslint, frontend, html, l'editorconfig, L'editorconfig, LeMonde, ok, phpunit, plugue, pre, psr, PSR, repo, répository, Scrutinizer, Speedcurve, Stylelint, stylelintrc, symfony, Symfony, sympas, technos, travis, trollLand, tutoriel, webperformance, WebPerformance

**Comment by ElevenWilson** [comment]
37 words might be misspelled, please check them: airbnb, BackstopJs, Bithound, captainjojo, config, CRON, css, CSS, editorconfig, Editorconfig, Eslint, frontend, html, l'editorconfig, L'editorconfig, LeMonde, ok, phpunit, plugue, pre, psr, PSR, repo, répository, Scrutinizer, Speedcurve, Stylelint, stylelintrc, symfony, Symfony, sympas, technos, travis, trollLand, tutoriel, webperformance, WebPerformance

**Comment by ElevenWilson** [comment]
37 words might be misspelled, please check them: airbnb, BackstopJs, Bithound, captainjojo, config, CRON, css, CSS, editorconfig, Editorconfig, Eslint, frontend, html, l'editorconfig, L'editorconfig, LeMonde, ok, phpunit, plugue, pre, psr, PSR, repo, Scrutinizer, Speedcurve, Stylelint, stylelintrc, suffsait, symfony, Symfony, sympas, technos, travis, trollLand, tutoriel, webperformance, WebPerformance

**Pull request merged**